addons/website_sale_collect/tests/common.py

```diff
@@ -3,26 +3,44 @@
 from odoo.fields import Command
 
 from odoo.addons.payment_custom.tests.common import PaymentCustomCommon
-from odoo.addons.website_sale.tests.common import WebsiteSaleCommon
+from odoo.addons.website_sale_stock.tests.common import WebsiteSaleStockCommon
 
 
-class OnSiteCommon(PaymentCustomCommon, WebsiteSaleCommon):
+class InStoreCommon(PaymentCustomCommon, WebsiteSaleStockCommon):
 
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
 
-        # Create the in-store delivery method.
-        product = cls._prepare_carrier_product(list_price=0.0)
-        cls.store_1 = cls.env['stock.warehouse'].create({
-            'name': 'Store 1',
-            'code': 'ST1',
-        })
-        cls.carrier = cls._prepare_carrier(
-            product,
+        cls.warehouse_1 = cls.env['stock.warehouse'].search([
+            ('company_id', '=', cls.env.company.id)
+        ])
+        cls.warehouse_2 = cls._create_warehouse()
+
+        cls.product_1 = cls._create_product()
+        cls._add_product_qty_to_wh(cls.product_1.id, 10, cls.warehouse_1.lot_stock_id.id)
+        cls.product_2 = cls._create_product()
```

Comment:
Seems like it could be in the `setUpClass` of `TestSaleOrder` :eyes:
Unless necessary, keep the test data for where they're needed, as it'll be created for all classes using the common.

For the second warehouse too, maybe it should only be in the test needing it :thinking:

Reply:
indeed, fixed!